### DISCUSSION TITLE
Add Server and Organization Info in Web hook Body Payload

### DISCUSSION THREAD

**Discussion by dipakparmar** [postalserver/postal#1928] [Feature suggestions]
Including the server permalink and organization slug into the webhook payload to allow users to directly access messages on the portal as other necessary information is already available in the payload like message-id. 

Adding this users can directly form the clickable link that will open the email in the browser. 

`DomainDNSError` has all the necessary variables available but not other like MessabeBounce or Held

Requesting adding following to other webhook events:

`:server => self.owner.webhook_hash,`

**Comment by willpower232** [comment]
If your application only has one Postal credential then you would be able to figure this out yourself because it would be the same for all messages?

**Reply by dipakparmar** [reply]
Hi, @willpower232 thanks for the reply. Though in production use normally users will have more than one mail server, and maybe more than one domain. If users have more than one mail server in one organization then they are definitely going to have more than one credential so it's a very rare case your suggestion would be helpful to many other users.

In our use case, we have more than one mail server in a single org to have proper isolation. Having the org and mailserver slug, make the complete URL to open the detected email in postal rather than finding it. 

Similarly to the [DNS Error Event](https://docs.postalserver.io/developer/webhooks#dns-error-event)

![Shot 2022-03-28 at 11 00 11@2x](https://user-images.githubusercontent.com/24366206/160458879-343b7e9a-ca0d-497f-aecf-169274b051e7.png)

**Comment by MexHigh** [comment]
Same problem here. I want to compose a URL for a message off of the message payload, so that the user can open the message info in the dashboard. However, the organisation slug and host is missing from the payload.

An alternative would be to just add a permalink directly to the webhook payload.
